lib/cell.rb

```diff
@@ -0,0 +1,45 @@
+require './lib/ship'
+
+class Cell
+    attr_reader :coordinate, :ship
+
+    def initialize(coordinate)
+        @coordinate = coordinate
+        @ship = nil
+        @fired_upon = false
+    end
+
+    def empty?
+        @ship == nil
+    end
+
+    def place_ship(ship)
+        @ship = ship
+    end
+
+    def fired_upon?
+      @fired_upon
+    end
+
+    def fire_upon
+      if @ship != nil
```

Comment:
```
def fire_upon
  @fired_upon = true
  @ship&.hit
```

The object&.method call will only call the method if the object is defined (i.e. not nil)